src/plugins/visualizations/public/vis_types/base_vis_type.ts

```diff
@@ -54,7 +54,7 @@ export class BaseVisType {
   stage: 'experimental' | 'beta' | 'production';
   isExperimental: boolean;
   options: Record<string, any>;
-  visualization: VisualizationControllerConstructor;
+  visualization: VisualizationControllerConstructor | undefined;
```

Comment:
nit: Maybe you could do `visualization?: VisualizationControllerConstructor;` instead of adding `undefined`?

Reply:
yeah i tried that, but it gets into some ts errors so i went with this